x-pack/plugins/watcher/public/sections/watch_edit/components/watch_edit.tsx

```diff
@@ -38,7 +39,12 @@ const watchReducer = (state: any, action: any) => {
     case 'setWatch':
       return payload;
     case 'setProperty':
-      return new (Watch.getWatchTypes())[state.type]({ ...state, ...payload });
+      const { property, value } = payload;
+      if (!isEqual(state[property], value)) {
```

Comment:
I would suggest flipping the logic here: `if (isEqual(state[property], value)) {` ...

Reply:
good point, fixed!